### DISCUSSION TITLE
Ropsten Testnet deprecated

### DISCUSSION THREAD

**Discussion by Michele971** [reach-sh/reach-lang#1452] [Q&A] [answered]
Hi guys,
Ropsten testnet has been deprecated so, when we run on ETH-live, how can we set a new Ethereum testnet?

Actually, If I try to run on Ropsten, default, the error is: 
`Error: bad response (status=410, headers={"server":"awselb/2.0","date":"Thu, 06 Oct 2022 07:23:16 GMT","content-type":"application/json; charset=utf-8","content-length":"253","connection":"close"}, body="{\n  \"id\":null,\n  \"error\": {\n    \"code\":-32601,\n    \"message\":\"Network decommissioned, please use Goerli or Sepolia instead\",\n    \"data\": {\n      \"see\":\"https://blog.infura.io/post/deprecation-timeline-for-rinkeby-ropsten-and-kovan-testnets\"\n    }\n  }\n}\n", requestBody="{\"method\":\"eth_blockNumber\",\"params\":[],\"id\":49,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="https://ropsten.infura.io/v3/84842078b09946638c03157f83405213", code=SERVER_ERROR, version=web/5.7.1)`

I tried to 'workaround' this issue using the command`REACH_CONNECTOR_MODE=ETH-live ETH_NODE_URI=https://goerli.infura.io/v3/[my_key]` but it didn't work. 

Any ideas?

Thanks in advance! :)

P.S.
Do you know any Ethereum testnet that has a good and efficient **faucet**?

**Comment by jeapostrophe** [comment, marked as answer]
You can connect to any network you want by setting these variables:

https://docs.reach.sh/networks/#p_60

Either in the environment or in the call to `loadStdlib`

**Reply by Michele971** [reply]
Yes, I checked the documentation but, even if I set the ETH_NODE_URI to a Goerli testnet node, it does not works.
Could you provide me an example? As I said, I used the following command: `REACH_CONNECTOR_MODE=ETH-live ETH_NODE_URI=https://goerli.infura.io/v3/[my_key]` but Reach said that I'm still using the Ropsten Testnet. I also tried using one of these [Goerli nodes](https://mudit.blog/getting-started-goerli-testnet/).

Thanks!

**Reply by jeapostrophe** [reply]
Are you also calling `setProviderByName('TestNet')`?

**Reply by Michele971** [reply]
Yes, I'm calling it! 
I don't know if it is important to you: I'm using RPC calls with python.

Now, when I wrote `ETH-live` which testnet am I using? Ropsten, right?

**Reply by jeapostrophe** [reply]
If you call that, then all default settings from the env are ignored. If you call setProviderByEnv then you can pass the rpc URL too

**Reply by Michele971** [reply]
Ok, got it! 
It works! I set `rpc("/stdlib/setProviderByEnv",{"ETH_NODE_URI":"https://goerli.infura.io/v3/[my_key]"})`.

Many thanks!

**Reply by jeapostrophe** [reply]
Phew!
